deploy/platform/kubernetes/feature-agent/resources.yaml

```diff
@@ -44,14 +44,23 @@ spec:
   template:
     metadata:
       labels:
+        # @feature: java-agent-injector; enable the java agent injector
+        swck-java-agent-injected: "true"
         app: gateway
       annotations:
         sidecar.istio.io/inject: "${AGENTLESS}"
+        # @feature: java-agent-injector; set the java agent configuration
+        strategy.skywalking.apache.org/inject.Container: "gateway"
+        strategy.skywalking.apache.org/agent.Overlay: "true"
+        agent.skywalking.apache.org/agent.service_name: "agent::gateway"
     spec:
       containers:
         - name: gateway
           image: ${HUB}/gateway-service:${TAG}
```

Comment:
@kezhenxu94 Is this image already agent-based? If so, it could confuse users.

Reply:
how about adding a full example of the operator, java agent injection would work smoothly in it.

Reply:
> how about adding a full example of the operator, java agent injection would work smoothly in it.

I think showcase is not all for demonstrating whole SkyWalking ecosystem works. Because we have too many components to do. Here, I am just asking to keep the logic simpler. We have the no-agent-included image, choosing that would be better.

Reply:
> @kezhenxu94 Is this image already agent-based? If so, it could confuse users.

`${TAG}` is substitutable, and it's substituted according to the feature deployed, when deploying agent it's `xxxxx` when it's agentless it's `xxxxx-agentless`

Reply:
Is there a way we use `agentless` image here?(For Java) Because we have injector already.

Reply:
> Is there a way we use `agentless` image here?(For Java) Because we have injector already.

It's already agentless, check this `TAG=${TAG}-agentless`,
 
https://github.com/apache/skywalking-showcase/blob/3ee8f9923e29c542996735464949fd002a31fa70/deploy/platform/kubernetes/features.mk#L57